app/models/project.rb

```diff
@@ -104,11 +104,25 @@ def aggregation_preference_allowed?
     end
   }
 
-  has_attached_file :icon, 
-    :styles => { :thumb => "48x48#", :mini => "16x16#", :span1 => "30x30#", :span2 => "70x70#", :original => "1024x1024>" },
-    :path => ":rails_root/public/attachments/:class/:attachment/:id/:style/:basename.:extension",
-    :url => "/attachments/:class/:attachment/:id/:style/:basename.:extension",
-    :default_url => "/attachment_defaults/general/:style.png"
+  if Rails.env.production?
+    has_attached_file :icon,
+      storage: :s3,
+      s3_credentials: "#{Rails.root}/config/s3.yml",
+      s3_protocol: CONFIG.s3_protocol || "https",
+      s3_host_alias: CONFIG.s3_host || CONFIG.s3_bucket,
+      bucket: CONFIG.s3_bucket,
+      styles: { thumb: "48x48#", mini: "16x16#", span1: "30x30#", span2: "70x70#", original: "1024x1024>" },
+      path: "projects/:id-icon-:style.:extension",
+      url: ":s3_alias_url",
+      default_url: "/attachment_defaults/general/:style.png"
+  else
```

Comment:
I'd also add in the CloudFront cache invalidator, which will send invalidation requests when people update their icon
```invalidate_cloudfront_caches :cover, "projects/:id-icon*"```